addons/website_forum/models/forum_post.py

```diff
@@ -276,27 +283,26 @@ def _search_can_view(self, operator, value):
 
         user = self.env.user
         # Won't impact sitemap, search() in converter is forced as public user
-        if self.env.is_admin():
-            return [(1, '=', 1)]
-
-        req = """
-            SELECT p.id
-            FROM forum_post p
-                   LEFT JOIN res_users u ON p.create_uid = u.id
-                   LEFT JOIN forum_forum f ON p.forum_id = f.id
-            WHERE
-                (p.create_uid = %s and f.karma_close_own <= %s)
-                or (p.create_uid != %s and f.karma_close_all <= %s)
-                or (
-                    u.karma > 0
-                    and (p.active or p.create_uid = %s)
-                )
-        """
-
-        op = 'inselect' if operator == '=' else "not inselect"
+        # not `env.is_admin` (not `su`) as this `can_view` field is used in ir.rules
+        if user._is_superuser() or user.user_has_groups(','.join(self._get_can_view_all_access_groups())):
+            return [(1, '=', 1) if operator == '=' else ('id', '=', 0)]
```

Comment:
See FALSE_LEAF in expression.py
```suggestion
            return [(1, '=', 1) if operator == '=' else (0, '=', 1)]